sourced/ml/mining/cmd/clickhouse2deps.py

```diff
@@ -0,0 +1,110 @@
+import json
+import logging
+import os
+
+from clickhouse_driver import Client
+import jinja2
+from scipy.sparse import coo_matrix
+
+from sourced.ml.mining.models import Dependencies
+
+
+QUERY_TEMPLATE = "clickhouse2deps.sql.jinja2"
+QUERY_ARGS = "clickhouse2deps.json"
+CLICKHOUSE_LANGS = [
+    "cpp",
+    "csharp",
+    "go",
+    "java",
+    "javascript",
+    "php",
+    "python",
+]  # TODO: add ruby
+
+
+def clickhouse2deps(args):
+    """
+    Extract dependencies from UASTs in a ClickhouseDB.
+    """
+    log = logging.getLogger("clickhouse2deps")
+    if not args.output_path.endswith(".asdf"):
+        args.output_path += ".asdf"
+    if os.path.exists(args.output_path):
+        if not args.force:
+            log.error(
+                "'%s' already exists, aborting (use force to remove).", args.output_path
+            )
+            raise FileExistsError
+        log.warn("'%s' already exists, removing.", args.output_path)
+        os.remove(args.output_path)
+    log.info("Loading the query template ...")
+    template_loader = jinja2.FileSystemLoader(
+        (os.path.dirname(__file__)), followlinks=True
+    )
+    env = jinja2.Environment(keep_trailing_newline=False)
+    template = template_loader.load(env, name=QUERY_TEMPLATE)
+    log.info("Loading the query args ...")
+    with open(os.path.join(os.path.dirname(__file__), QUERY_ARGS), "r") as fin:
+        query_args = json.load(fin)
+    client = Client(
+        user=args.user,
+        password=args.password,
+        host=args.host,
+        port=args.port,
+        database=args.database,
+    )
+    files, deps, ind_files, ind_deps = [], [], [], []
+    ind_to_langs, ind_to_repos, file_to_inds, dep_to_inds = {}, {}, {}, {}
+    for lang in args.langs:
+        log.info("Extracting the dependencies of %s files ...", lang)
+        rows = client.execute_iter(
+            template.render(lang=lang, table=args.table, query_args=query_args[lang]),
+            settings={"max_block_size": args.max_block_size},
+        )
+        num_rows, num_files, num_deps = 0, 0, 0
+        for row in rows:
+            try:
+                row = [e.decode("utf-8") if not isinstance(e, str) else e for e in row]
```

Comment:
```suggestion
                row = [e.decode("utf-8", errors="ignore") if not isinstance(e, str) else e for e in row]
```

thus no need to catch unicode errors

Reply:
Ah nice